Start arch doc (#1556)
ActiveState/cli: architecture.md

Tool Applications.

## Directory Structure

### assets/

Various files (e.g. images, templates, etc.) used by any application.

### build/

Artifacts resulting from building applications.

### cmd/*/

Individual "main" applications.

#### cmd/*/internal/

Packages used exclusively for the parent application.

### docs/

Developer-focused documentation.

### .github/workflows-src/

YAML-formatted config that is processed by `ytt`
(https://github.com/vmware-tanzu/carvel-ytt) to produce yml files used for CI
(stored in `.github/workflows/`).

### installers/

Shell scripts wrapping the installer application for user convenience.

### internal/

Packages that are made available for use by any application, but are restricted
from use by external code.

#### internal/runbits/

Packages that are made available for use by "runner" packages. In essence,
`internal/runners/internal/runbits`.

#### internal/runners/

Packages that provide command behavior for the State Tool application.

### locale/

Localization keys and associated values.

### pkg/

Packages that are made available for use by any application, including external

> code. The packages that are currently located here are not intended for use by
> external code, and will eventually be moved into the `internal/` directory.

Reviewer: ```suggestion
code. Note we are changing our strategy with regards to these packages, 
and these will eventually be moved into the `internal/` directory.
```

#### pkg/cmdlets/

Packages that are made available for use by "runner" packages. Synonymous with

> `internal/runbits/`, and all new runner-common packages should be placed there.

Reviewer: Other way around; runbits is new, cmdlets are old.

Author: That is what I tried to convey, that runbits is the location of all new common packages. Should I word it differently? Maybe:
```suggestion
#### pkg/cmdlets/

Packages that are made available for use by "runner" packages. Synonymous with
`internal/runbits/` with all new runner-common packages being placed there.
```

Reviewer: I think you should make the title `pkg/runbits` and instead hint at cmdlets. runbits are the way forward, so it should take center stage.

Author: That directory is covered: https://github.com/ActiveState/cli/pull/1556/files#diff-4e03c430fc7f575c72fd989625acdc7d8d70fd20d3693c893d67d9dbd461ccdbR77-R81

Author: Just pushed a change to remove the pkg/cmdlets entry and update the runbits entry.

#### pkg/platform/

> Packages focused on interacting with the platform. Much of the behavior is

Reviewer: ```suggestion
Packages focused on interacting with platform API's. Much of the behavior is
```

generated, but there are also critical components providing platform-dependent
client-side logic (e.g. `pkg/platform/runtime`).

#### pkg/{project,projectfile}/

Packages that provide setup and interaction with the activestate.yaml files.

### scripts/

> Helper scripts.

Reviewer: ```suggestion
Helper scripts for development and deployment processes.
```
